posthog/api/event.py

```diff
@@ -225,19 +225,33 @@ def distribution(query):
                         FROM (SELECT global_session_id, EXTRACT(\'EPOCH\' FROM (MAX(timestamp) - MIN(timestamp)))\
                             AS length FROM ({}) as count GROUP BY 1) agg'.format(query)
 
-        result = []
+        def average_length_time(query):
+            return 'SELECT date_trunc(\'day\', timestamp) as start_time,\
+                        AVG(length) AS average_session_length\
+                        FROM (SELECT global_session_id, EXTRACT(\'EPOCH\' FROM (MAX(timestamp) - MIN(timestamp)))\
+                            AS length,\
+                            MIN(timestamp) as timestamp FROM ({}) as count GROUP BY 1) as agg group by 1 order by start_time'.format(query)
+
+        result: List = []
         if session_type == 'avg':
             cursor = connection.cursor()
             cursor.execute(overall_average_length(all_sessions), sessions_sql_params)
             calculated = cursor.fetchall()
-            avg_length = round(calculated[0][1], 0)
+            avg_length = round(calculated[0][1], 0) if calculated[0][1] is not None else 0
             avg_formatted = friendly_time(avg_length)
-            result = [{'label': 'Number of Sessions', 'count': calculated[0][0]}, {'label': 'Average Duration of Session', 'count': avg_formatted}]
+            overall_average = {'label': 'Average Duration of Session', 'count': avg_formatted}
+
+            cursor = connection.cursor()
+            cursor.execute(average_length_time(all_sessions), sessions_sql_params)
```

Comment:
These queries aren't cheap. I wonder if you could combine them into one query, maybe by getting both the average and the number of sessions for each day and then summing that. 

Also, in `overall_average_length`, the query might be a bit quicker if you do `SELECT COUNT(1)` instead of `COUNT(*)`.

Reply:
Condensed this so that it calculates the average outside the query and only has to use one query to get the time series data and totals. This could be a different slow down but I think calculating an average serverside rather than submitting a whole query for that one number is better. Though, this hasn't been clocked so I'm not completely sure.

Reply:
It's still a bit slow (50s with a few million events) though no more so than before